modules/util/create.py

```diff
@@ -929,9 +929,8 @@ def create_optimizer(
                 else:
                     # the group state was not saved, initialize with an empty group state
                     new_group = new_param_groups[new_group_index]
-                    for i in range(len(new_group['params'])):
-                        new_group['params'][i] = state_index
-                        state_index += 1
+                    new_group['params'] = list(range(state_index, state_index + len(new_group['params'])))
```

Comment:
I've verified that this is equivalent to the old code. Good job refactoring it!

Reply:
It's only the same if the original value of `new_group['params']` has the same size as `new_group['params']`, which is something we can't assume to be true

Reply:
Darn, you're right @Nerogar 

It overwrites the exact same indices and creates them with the same values as before. But if `new_group['params']` has any more indices AFTER that, the new code erases them. I just assumed that was what we wanted to do, but you're right, it looks like we need to go back to the oldest implementation here:

```py
                    for i, old_state_index in enumerate(new_group['params']):
                        new_group['params'][i] = state_index
                        state_index += 1
```

Or if we needed to fix some ruff lint about unused variables, we can use the first rewrite from this PR:

```py
                    for i in range(len(new_group['params'])):
                        new_group['params'][i] = state_index
                        state_index += 1
```

Reply:
Well, the loop is over `new_group['params']`, and the `range` is from `state_index` to `state_index + len(new_group['params'])`, which comes to the same length, so I'm not sure what you mean there.

Reply:
Damn, all this time I misread it as "create a new list from somewhere else, then loop through an old list and update the new list at the same indices".

It turns out it's actually "make a new list, then loop over all of its own elements and update them".

So there is no bug in the new code at all. It's only modifying the exact same list with the exact same result as before the refactoring.

Reply:
You are right. I shouldn't read when I'm too tired. I also thought it was modifying a different list.